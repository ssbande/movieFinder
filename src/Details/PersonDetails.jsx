import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Header, Image } from 'semantic-ui-react';

import { createImageUrl } from '../utils/helpers';

const PersonDetail = ({ detailInfo }) => {
  const { profile_path, name, biography, place_of_birth, birthday } = detailInfo;

  return <Grid stackable stretched>
    <Grid.Column width={5}>
      <Image className="posterImage" src={createImageUrl(profile_path)} size='medium' bordered />
    </Grid.Column>
    <Grid.Column width={11}>
      <div>
        <Header size='huge' color='yellow' inverted dividing>{name}</Header>
        <div>{biography}</div>
        <Header as='h3' inverted color='grey'>
          Place of birth
          <Header.Subheader>{birthday} {place_of_birth}</Header.Subheader>
        </Header>
      </div>
    </Grid.Column>
  </Grid>
}

PersonDetail.propTypes = {
  detailInfo: PropTypes.shape({
    profile_path: PropTypes.string,
    name: PropTypes.string,
    biography: PropTypes.string,
    place_of_birth: PropTypes.string,
    birthday: PropTypes.string
  })
}

export default PersonDetail;