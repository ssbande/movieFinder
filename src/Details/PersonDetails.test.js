import React from 'react';
import { shallow } from 'enzyme';

import PersonDetails from './PersonDetails';

const setup = () => ({
  props: {
    detailInfo: {
      id: 1001,
      profile_path: 'profile_path',
      name: 'name',
      biography: 'biography',
      place_of_birth: 'place_of_birth',
      birthday: '1990-01-01'
    }
  }
})

describe('Person Detail', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<PersonDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})