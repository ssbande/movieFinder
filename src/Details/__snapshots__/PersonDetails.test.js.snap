// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Detail should render correctly 1`] = `ShallowWrapper {}`;
